### PR TITLE
OPS-8396 Add LB logging to S3

### PR DESCRIPTION
**Pull request by daleckystepan**

### CODE REVIEW

File: aws/load-balancer.tf [hunk 1]
```diff
@@ -43,6 +43,26 @@ resource "aws_lb_listener" "rancher_https" {
   }
 }
 
+resource "aws_s3_bucket" "rancher_lb_access_logs" {
+  bucket        = "wandera-${var.account_name}-rancher-access-logs"
```
Comment (gerricchaplin) [inline aws/load-balancer.tf]: Please make this optional using a count on var length so we can switch on and off from the top level.

File: aws/load-balancer.tf [hunk 2]
```diff
@@ -51,5 +71,10 @@ resource "aws_lb" "rancher_lb" {
   load_balancer_type = "application"
   idle_timeout       = "${var.alb_idle_timeout}"
 
+  access_logs {
+    bucket          = "${aws_s3_bucket.rancher_lb_access_logs.bucket}"
+    enabled         = "1"
```
Comment (gerricchaplin) [inline aws/load-balancer.tf]: Please make this optional using a count on var

File: aws/variables.tf [hunk 3]
```diff
@@ -1,3 +1,12 @@
+variable "account_name" {
+  type        = "string"
+  description = "AWS account name"
+}
+
+variable "environment" {
+  type    = "string"
+  default = "operations-shared"
+}
```
Comment (gerricchaplin) [inline aws/variables.tf]: Please allow for passing a variable to enable/disable creation fo S3 resources and ALB logging. Default false.

File: aws/load-balancer.tf (hunk 1 above)
Comment (gerricchaplin) [reply aws/load-balancer.tf]: This should not have the word Wandera in it as this is OSS. Please allow passing bucket name in from top level.

File: aws/variables.tf [hunk 4]
```diff
@@ -1,3 +1,7 @@
+variable "environment" {
+  type    = "string"
```
Comment (gerricchaplin) [inline aws/variables.tf]: Shouldn't need this because of var.cloud_tags

File: aws/load-balancer.tf [hunk 5]
```diff
@@ -43,6 +43,27 @@ resource "aws_lb_listener" "rancher_https" {
...
+
+    expiration {
+      days = 10
+    }
+  }
+
+  tags {
+    Name        = "Access logs for Rancher loadbalancer"
+    Environment = "${var.environment}"
+    env         = "${var.environment}"
```
Comment (gerricchaplin) [inline aws/load-balancer.tf]: Name is usually a noun. `Description` is probably what you are looking for.
You may want to use something like the following so as to get default tags being passed in which include env and Environment.
`tags = "${merge(map("Name", "rancher"), var.cloud_tags)}"`

File: aws/load-balancer.tf [hunk 6]
```diff
@@ -43,6 +43,23 @@ resource "aws_lb_listener" "rancher_https" {
...
+
+  lifecycle_rule {
+    enabled = true
+
+    expiration {
+      days = 10
+    }
+  }
+
+  tags = "${merge(map("Name", "rancher"), var.cloud_tags)}"
```
Comment (czernitko) [inline aws/load-balancer.tf]: The bucket name should probably not be hardcoded. Using `var.rancher_lb_access_logs_bucket` should be fine. Or if hardcoded, then at least `rancher-lb-access-logs` should be used, but full bucket name is preferable.

**Review by czernitko — APPROVED** [review]

**Review by czernitko — APPROVED** [review]

**Review by gerricchaplin — APPROVED** [review]
LGTM

**Pull request merged**